ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/PieceCrafterDetailPanel.java

```diff
@@ -178,44 +193,45 @@ private void initComponents() {
             data.list.addListSelectionListener(e -> refreshButtonState(data));
         });
 
-        JPanel boardAndSave = new JPanel();
-        boardAndSave.setOpaque(false);
-        boardAndSave.setLayout(new GridBagLayout());
+        JPanel boardPanels = new JPanel();
+        boardPanels.setOpaque(false);
+        boardPanels.setLayout(new BoxLayout(boardPanels, BoxLayout.LINE_AXIS));
+        boardPanels.add(Box.createHorizontalGlue());
+        boardPanels.add(mBoardPanel);
+        boardPanels.add(Box.createHorizontalGlue());
+        boardPanels.addComponentListener(new ComponentListener() {
+            @Override
+            public void componentResized(ComponentEvent e) {
+                mBoardPanel.updateDimensions(e.getComponent().getWidth(), e.getComponent().getHeight());
+                mBoardPanel.revalidate();
```

Comment:
Can `BoardPanel#updateDimensions()` just call `revalidate()` and `repaint()` internally, so we don't have to remember to call it everyplace else?